src/components/Button/Button.tsx

```diff
@@ -88,21 +88,19 @@ const Button = <P,>({
     {
       "has-icon": hasIcon,
       "is-dense": dense,
-      "is-disabled": Component !== "button" && disabled,
+      "is-disabled": disabled,
       "is-inline": inline,
       "is-small": small,
     },
     className
   );
   const onClickDisabled = (e: MouseEvent) => e.preventDefault();
-  const disabledProp =
-    Component === "button" ? { disabled } : { "aria-disabled": disabled };
 
   return (
     <Component
       className={classes}
       onClick={disabled ? onClickDisabled : onClick}
-      {...disabledProp}
+      aria-disabled={disabled ? disabled : undefined}
```

Comment:
Can this not just be `disabled` passed directly in? It's already a boolean/undefined value

Reply:
This is to explicitly skip this prop when false. This is similar to existing behaviour where `disabled={false}` results in `disabled` attribute not being rendered at all. I did this to reduce the impact of this change (such as failing snapshot tests).